dev/src/write-batch.ts

```diff
@@ -594,6 +594,29 @@ export class WriteBatch implements firestore.WriteBatch {
     });
   }
 
+  /**
+   * Removes all operations from the WriteBatch except for the ones located
+   * at the provided indexes.
+   *
+   * @param indexes List of operation indexes to keep
+   * @private
+   */
+  _sliceIndexes(indexes: number[]): void {
```

Comment:
The Array `slice()` method returns a copy, which I think would be preferably here as well.  Can we update this to create a new WriteBatch that is marked '_committed=false`? The reason I am asking is that I would ideally not expose a method that mocks with the internals of a WriteBatch, as the potential for future misuse is high.